source/vision/visionHandler.py

```diff
@@ -71,33 +127,43 @@ def postGuiInit(self) -> None:
 		self.handleConfigProfileSwitch()
 		config.post_configProfileSwitch.register(self.handleConfigProfileSwitch)
 
-	def terminateProvider(self, providerName: str) -> bool:
+	def terminateProvider(

```

Comment:
Hmm. I was trying to keep a consistent interface. Rather than using `providerId` as the token for a provider, using `ProviderInfo`. Another option which might get around a few of these issues is to have the `VisionHandler` class maintain a dictionary of `providerId` to `ProviderInfo` objects. This way, when it needs to know the associated module name etc it is easy to look up. We would then revert these methods to take `providerId`, perhaps add one to `getProviderInfo` from a `providerId`. Thoughts?

Reply:
I think that makes sense.

Reply:
No, looking at this again, I'd prefer to keep this functions using `providerId`. All the places that use this already have access to and need the full `providerInfo`. I think this approach is more extensible, if these methods need some other information about a provider it is easy to include. For the python console I think it reasonable to do:
```
import vision
vision.handler.terminateProvider(vision.visionHandler.getProviderInfo("screenCurtain"))
```
Though I think I will move a this around slightly so that it is instead:
```
import vision
vision.handler.terminateProvider(vision.handler.getProviderInfo("screenCurtain"))
```

Reply:
Sounds good.